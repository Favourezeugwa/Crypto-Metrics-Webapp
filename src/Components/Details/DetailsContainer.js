import React from 'react';
import { Container } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import './Details.css';

function DetailsContainer() {
  const cryptos = useSelector((state) => state.crypto);

  const filteredCrypto = cryptos.filter((crypto) => crypto.show === true);

  return (
    <Container>
      <div className="detailsContainer">
        {
         filteredCrypto.map((crypto) => (
           <>
             <div className="imageContainer">
               <img src={crypto.img} alt="coin logo" id={crypto.id} className="indiImg" />
               <div className="cont">
                 <h2 key={crypto.id}>
                   Coin:
                   {' '}
                   {crypto.name}
                 </h2>
                 <h2 key={crypto.id}>
                   Ranked at:
                   {' '}
                   #
                   {crypto.rank}
                 </h2>
               </div>
             </div>
             <h2 className="detailTitle" key={crypto.id}>
               More Information about
               {' '}
               {crypto.name}
             </h2>
             <div className="info">
               <p>Coin Name</p>
               <p key={crypto.id}>{crypto.name}</p>
             </div>
             <div className="info">
               <p>Ranking</p>
               <p key={crypto.id}>
                 No.
                 {crypto.rank}
               </p>
             </div>
             <div className="info">
               <p>Current Price</p>
               <p key={crypto.id}>
                 $
                 {crypto.price}
               </p>
             </div>
             <div className="info">
               <p>Symbol</p>
               <p key={crypto.id}>{crypto.symbol}</p>
             </div>
             <div className="info">
               <p>Market Cap</p>
               <p key={crypto.id}>
                 $
                 {crypto.marketCap}
               </p>
             </div>
             <div className="info">
               <p>Available Supply</p>
               <p key={crypto.id}>{crypto.availableSupply}</p>
             </div>
             <div className="info">
               <p>Contract Address</p>
               <p key={crypto.id}>{crypto.contractAddress}</p>
             </div>
             <div className="info">
               <p>Decimals</p>
               <p key={crypto.id}>{crypto.decimals}</p>
             </div>
             <div className="info">
               <p>Price Btc</p>
               <p key={crypto.id}>
                 $
                 {crypto.priceBtc}
               </p>
             </div>
             <div className="info">
               <p>Price Change/1 Day Ago</p>
               <p key={crypto.id}>
                 $
                 {crypto.priceChange1d}
               </p>
             </div>
             <div className="info">
               <p>Price Change/Hour</p>
               <p key={crypto.id}>
                 $
                 {crypto.priceChange1h}
               </p>
             </div>
             <div className="info">
               <p>Price Change/Week</p>
               <p key={crypto.id}>
                 $
                 {crypto.priceChange1w}
               </p>
             </div>
             <div className="info">
               <p>Total Supply</p>
               <p key={crypto.id}>
                 $
                 {crypto.totalSupply}
               </p>
             </div>
             <div className="info">
               <p>Volume</p>
               <p key={crypto.id}>{crypto.volume}</p>
             </div>
             <div className="infoLink">
               <p>Twitter</p>
               <li><a href={crypto.twitterUrl} key={crypto.id}>Our Twitter Handle</a></li>
             </div>
             <div className="infoLink">
               <p>Website</p>
               <li><a href={crypto.websiteUrl} key={crypto.id}>Our website URL</a></li>
             </div>
           </>
         ))
        }
      </div>
    </Container>
  );
}
export default DetailsContainer;
