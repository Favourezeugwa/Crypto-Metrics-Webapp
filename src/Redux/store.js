import { configureStore } from '@reduxjs/toolkit';
import CryptoReducer from './Crypto/CryptoSlice';

export default configureStore({
  reducer: {
    crypto: CryptoReducer,
  },
});
