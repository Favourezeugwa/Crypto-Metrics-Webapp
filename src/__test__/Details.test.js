import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import DetailsContainer from '../Components/Details/DetailsContainer';
import store from '../Redux/store';

it('Test DetailsContainer renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <DetailsContainer />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
