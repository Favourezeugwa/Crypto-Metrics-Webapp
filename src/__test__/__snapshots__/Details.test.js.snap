// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DetailsContainer renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="detailsContainer"
  />
</div>
`;
